cmd/dagger/engine.go

```diff
@@ -3,18 +3,26 @@ package main
 import (
 	"context"
 
+	"dagger.io/dagger/telemetry"
 	"github.com/dagger/dagger/engine"
 	"github.com/dagger/dagger/engine/client"
-	"github.com/dagger/dagger/telemetry"
+	"github.com/moby/buildkit/identity"
 )
 
 type runClientCallback func(context.Context, *client.Client) error
 
+// ClientID is the client ID for the CLI's connection to the engine.
+//
+// It must be established as soon as possible for telemetry purposes.
+var ClientID = identity.NewID()
```

Comment:
The new draining logic is based on client IDs, so now we determine the client ID even earlier than connecting to the engine, so we can immediately annotate telemetry coming out of the CLI

Reply:
In the case of a nested dagger CLI the client ID ends up being the one set by the executor, so this would not be used. Guessing that's fine cause we [short-circuit that case](https://github.com/sipsma/dagger/blob/0b36cdd745611b75b43f13ef6158de5a45d4bd00/engine/client/client.go#L145-L145) in the client anyways, just double checking my overall understanding

Reply:
Good call, ended up removing this, it wasn't load-bearing and I couldn't really come up with a good reason to keep it besides "having a client set everywhere felt good." It doesn't end up being set everywhere anyway, and it doesn't need to, since the full span ancestry chain is walked to collect client IDs.

Would rather just not have it than have misleading values in telemetry for the nesting case. :+1: